### ISSUE TITLE
SENTRY_USE_X_FORWARDED_FOR not working

### ISSUE THREAD

**Issue by AndreKR** [getsentry/sentry#22955]
## Important Details

How are you running Sentry?

On-Premise w/ Docker, version 20.11.1

## Description

When a CSP report comes in, Sentry uses the IP of the Docker host (reverse proxy) as the client IP. It ignores the `X-Forwarded-For` header.

Obviously honoring the `X-Forwarded-For` header is dangerous and must be enabled only when a reverse proxy is used. (But with the Sentry On-premise Docker installation almost *always* a reverse proxy is used.) It is unclear from the docs how it should be enabled.

I tried setting `SENTRY_USE_X_FORWARDED_FOR` in `.env` and it shows up in Admin->Environment:

![image](https://user-images.githubusercontent.com/1188538/103886170-63f81b00-50e1-11eb-8ca7-24ecb1a7dff6.png)

But still, events are reported with the IP of the Docker host:

![image](https://user-images.githubusercontent.com/1188538/103886338-a91c4d00-50e1-11eb-8991-a848093875d4.png)

## Steps to Reproduce

1. Install Sentry
2. Configure a reverse proxy of your choice on the docker host, forwarding requests to the docker container
3. Receive a CSP report (or an event from a Javascript client)

### What you expected to happen

To see the IP address of the browser, not of the reverse proxy.

### Possible Solution

I'm not sure why it is not working. I added another config option `SENTRY_SOME_OTHER_OPTION` to `.env` and it does *not* show up in Admin->Environment, so I am sure that I spelled SENTRY_USE_X_FORWARDED_FOR correctly and that the value actually reaches Sentry. The problem is just that it gets ignored apparently.

**Comment by AndreKR** [comment]
Still valid.

**Comment by BYK** [comment]
Will check soon-ish but no promises. If you can dive into the code and get a PR up, happy to review that (it would be faster than me getting to this probably).

**Comment by ZuSe** [comment]
Same problem here with running on k8s.

**Comment by Wirone** [comment]
I think it's related to guzzle/psr7#430 - default `RequestFetcher` used in `RequestIntegration` uses `GuzzleHttp\Psr7\ServerRequest::fromGlobals()` which does not support HTTP proxies.

**Comment by AndreKR** [comment]
@Wirone That issue is about something PHP while Sentry is Python, highly unlikely that it has anything to do with this issue.

**Comment by Wirone** [comment]
@AndreKR yeah, maybe it's not best place for my comment, it should be rather in `getsentry/sentry-php` because PHP SDK does not handle proxies properly. I did not read description properly, saw keywords and thought it's related 😉
